### PR TITLE
oav2.3.1 dependency analyze. auto genereate test-scenario. run test scenario

### PR DESCRIPTION
**Pull request by ruowan**
scenario

### CODE REVIEW

**Review by PhoenixHe-msft — APPROVED** [review]

File: lib/testScenario/gen/staticTestScenarioGenerator.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+          ),
+        },
+      ];
+      if (deleteExampleFilePath !== "") {
+        testStep.push({
+          step: deleteExampleName,
+          exampleFile: deleteExampleFilePath,
+        });
+      }
+      testDef.testScenarios.push({
```
Comment (raych1) [inline lib/testScenario/gen/staticTestScenarioGenerator.ts:102]: Do we use same test scenario to store different PUT+DELETE cases when there're multiple examples for PUT operation?

File: lib/testScenario/swaggerAnalyzer.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+  filerTopLevelResourceType?: boolean;
+}
+
+export interface ExampleDependency {
+  exampleFilePath: string;
+  exampleName: string;
+  apiVersion: string;
+  resourceProvider: string;
+  resourceType: string;
+  internalDependency: Dependency;
```
Comment (raych1) [inline lib/testScenario/swaggerAnalyzer.ts:40]: Should this be array too?

Comment (ruowan) [reply lib/testScenario/swaggerAnalyzer.ts:40]: internalDependency is analyzed by path. So it should be only one dependencyChain.

File: lib/testScenario/gen/staticTestScenarioGenerator.ts (hunk 1 above)
Comment (ruowan) [reply lib/testScenario/gen/staticTestScenarioGenerator.ts:102]: No. For multiple examples, we store it in separrate test scenario

**Pull request merged**